packages/cli/lib/auth.ts

```diff
@@ -43,62 +43,50 @@ const getMetadata = once(async (config: AuthConfig) => {
   return (await response.json()) as Metadata
 })
 
-const tokenGetter = function (config: AuthConfig, log: Debugger) {
-  let token: LiveToken | null = null
+const getToken = async function (config: AuthConfig, log: Debugger) {
+  const metadata = await getMetadata(config)
 
-  return async function getToken (): Promise<LiveToken> {
-    if (!token || !isValid(token)) {
-      const metadata = await getMetadata(config)
-
-      const params = {
-        'grant_type': 'client_credentials',
-        'client_id': config.clientId,
-        'client_secret': config.clientSecret,
-      }
-
-      config.params.forEach((value, key) => (params[key] = value))
+  const params = {
+    'grant_type': 'client_credentials',
+    'client_id': config.clientId,
+    'client_secret': config.clientSecret,
+  }
 
-      const response = await fetch(metadata.token_endpoint, {
-        method: 'POST',
-        headers: {
-          'Content-Type': 'application/x-www-form-urlencoded',
-        },
-        body: querystring.stringify(params),
-      })
+  config.params.forEach((value, key) => (params[key] = value))
 
-      const newToken = (await response.json()) as TokenResponse
+  const response = await fetch(metadata.token_endpoint, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/x-www-form-urlencoded',
+    },
+    body: querystring.stringify(params),
+  })
 
-      if ('error' in newToken) {
-        throw new Error(newToken.error_description || newToken.error)
-      }
+  const newToken = (await response.json()) as TokenResponse
 
-      log('Renewed access token', newToken)
+  if ('error' in newToken) {
+    throw new Error(newToken.error_description || newToken.error)
+  }
 
-      const expiration = Date.now() + newToken.expires_in * 1000
-      token = {
-        ...newToken,
-        expiration,
-      }
-    }
+  log('Renewed access token', newToken)
 
-    return token
+  const expiration = Date.now() + newToken.expires_in * 1000
+  return {
+    ...newToken,
+    expiration,
   }
 }
 
-export async function setupAuthentication (config: AuthConfig, log: Debugger) {
-  const getToken = tokenGetter(config, log)
+export function setupAuthentication (config: AuthConfig, log: Debugger) {
+  let token: LiveToken
```

Comment:
I'd be just a bit more specific with types by doing something like
```js
let token: LiveToken | null = null;
Hydra.defaultHeaders = async () => {
  if (!token || !isValid(token)) {
    // …
  }
}
```

and remove the null check in `isValid`

Reply:
Might as well do that. I initially tried to keep it that way but did something wrong and TypeScript wasn't happy. Second time's a charm tho